enhancements/network/network-policy-audit-logging.md

```diff
@@ -0,0 +1,315 @@
+      ipsecConfig: {}
+      aclLoggingRateLimit: 20
+```
+
+### User Stories
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+Include a story on how this proposal will be operationalized:  lifecycled, monitored and remediated at scale.
+
+#### Story 1
+
+As an Openshift user I want to see what Network Policies and accompanied ACLs are dropping the most traffic in a cluster. For example, when a packet runs into a `drop` ACL it will log a message similar to the following 
+
+```
+2021-01-05T17:34:02.675Z|00004|acl_log(ovn_pinctrl0)|INFO|name="<Network Policy Name>", verdict=drop, severity=info: icmp,vlan_tci=0x0000,dl_src=50:54:00:00:00:02,dl_dst=50:54:00:00:00:01,nw_src=192.168.0.3,nw_dst=192.168.0.2,nw_tos=0,nw_ecn=0,nw_ttl=64,icmp_type=8,icmp_code=0
+```
+
+#### Story 2
+
+As an Openshift user I want to control the rate of ACL logging to ensure I can extract the useful information invloving accept/reject actions in high hit rate scenarios. 
+
+#### Story 3 
+
+As an Openshift user I want to monitor in realtime traffic that is hitting any network policies I have implemented. 
+
+--- TBD in implementation (Realtime Monitering would be nice here)
+
+### Implementation Details/Notes/Constraints 
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+The following codebase changes will need to done before this feature becomes usable  
+
+- OVN Kubernetes 
+
+  - Move from a cluster wide namespace default deny port group to namespace specific one 
+  - Add logging flag and custom OVN meter to all created ACLs, see below 
+  ```
+  fmt.Sprintf("log=%v", aclLogging != ""),
+  fmt.Sprintf("severity=%s", getACLLoggingSeverity(aclLogging)), "meter=acl-logging",
+  ```
+- Cluster Network Operator 
+  
+  - Digest the `aclLoggingRateLimit` Flag and start OVN-K with the configured value 
+
+
+- Logging Stack 
+  
+  - We need a way to collect and present ACL logging from the OVN controller logs 
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+- How do we control who is able to see the audit logs. 
+   - Only Admins should be able to see the audit logs since it could help expose holes in a cluster's network policy structure 
+
+- Scale questions involving the addition of OVN meters to all created ACL's 
+  - Adding meters to each ACL does add a slightly larger overhead in OVN
```

Comment:
These two bullets could probably be one:
"Adding meters to each ACL does add a slightly larger overhead in OVN, but based on early upstream testing with logging both enabled and disabled, there were negligible performance impact detected."